lib/cisco_node_utils/fabricpath_global.rb

```diff
@@ -0,0 +1,439 @@
+#
+# NXAPI implementation of Fabricpath Global class
+#
+# November 2015, Deepak Cherian
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # node_utils class for fabricpath_global
+  class FabricpathGlobal < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      fail TypeError unless name.is_a?(String)
+      fail ArgumentError unless name == 'default'
+      @name = name.downcase
+      @set_params = {}
+
+      create if instantiate
+    end
+
+    def self.globals
```

Comment:
In't the code in  def self.fabricpath_feature_set(fabricpath_set) identical to that?


From: Chris Van Heuveln <notifications@github.com<mailto:notifications@github.com>>
Reply-To: cisco/cisco-network-node-utils <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, November 24, 2015 at 8:12 AM
To: cisco/cisco-network-node-utils <cisco-network-node-utils@noreply.github.com<mailto:cisco-network-node-utils@noreply.github.com>>
Cc: Cisco Employee <dcherian@cisco.com<mailto:dcherian@cisco.com>>
Subject: Re: [cisco-network-node-utils] Collapse feature/fabricpath into develop (#102)


In lib/cisco_node_utils/fabricpath_global.rb<https://github.com/cisco/cisco-network-node-utils/pull/102#discussion_r45756052>:

> +
> +module Cisco
> +  # node_utils class for fabricpath_global
> +  class FabricpathGlobal < NodeUtil
> +    attr_reader :name
> +
> +    def initialize(name, instantiate=true)
> +      fail TypeError unless name.is_a?(String)
> +      fail ArgumentError unless name == 'default'
> +      @name = name.downcase
> +      @set_params = {}
> +
> +      create if instantiate
> +    end
> +
> +    def self.globals


In addition -- considering that fabricpath is installed as a feature-set in a two-step process -- take a look at feature-set fex enablement to see how we've handled that:

https://github.com/cisco/cisco-network-node-utils/blob/develop/lib/cisco_node_utils/interface.rb#L127

-
Reply to this email directly or view it on GitHub<https://github.com/cisco/cisco-network-node-utils/pull/102/files#r45756052>.